### ISSUE TITLE
Is there a way to process this faster?

### ISSUE THREAD

**Issue by aeromomo** [mholt/PapaParse#316]
Now I have about 30-100MB/s to process.
Is there a way I have utilize Muti-Core and pipe on Nodejs?

**Comment by evanplaice** [comment]
Not a contributor but I have experience parsing CSV.

This is possible but likely very difficult. To stream parse CSV it would make sense to detect a predefined number of lines and pass them off to a worker process, repeat that step for N workers; and join the results at the end.

The issue is, the data itself can also contain newline chars. Ie, it's impossible to detect where line boundaries exist without parsing the actual data first.

Writing a CSV stream parser requires loading a chunk (ie up to a pre-defined char boundary), parsing up to that boundary, then loading the next chunk.

If you had a parser that just parses lines, this could work. For example, split the data into 1000 line chunks, send each subset to a worker that performs a second pass to parse out the values. The problem is, the second pass makes the parser 2x less efficient for users who don't use background workers. In addition, the process of concatenating the results incurs an additional performance penalty.

In short, it's possible but very difficult. Any benefits to parsing very large data sets would incur a noticeable performance penalty to users who aren't parsing very large dsta sets.

**Comment by pokoli** [comment]
I'm closing the issue as I think there is nothing to do in paparse. 

Thank you both for your comments.
